src/main/java/com/uid2/shared/auth/Roles.java

```diff
@@ -10,7 +10,9 @@
 
 public class Roles {
     public static <T extends Enum<T>> Set<T> getRoles(Class<T> type, JsonObject clientKeyJson) {
-        return getRoles(type, clientKeyJson.getJsonArray("roles", new JsonArray()));
+        return clientKeyJson.containsKey("roles")
+                ? getRoles(type, clientKeyJson.getJsonArray("roles"))
```

Comment:
Not critical here, but this scans the json object twice. If you are willing to break this one-liner you can avoid the additional pass:
```
JsonArray rolesArr = clientKeyJson.getJsonArray("roles");
return getRoles(type, rolesArr != null ? rolesArr : new JsonArray());
```